src/main/javatests/com/google/summit/translation/SourceLocationTest.kt

```diff
@@ -116,12 +115,38 @@ class SourceLocationTest {
             = 'Hello';
         }
         """.trim()
-    val loc = SourceLocation(startLine = 1, startColumn = 4, endLine = 3, endColumn = 16)
+    val loc = SourceLocation(startLine = 1, startColumn = 4, endLine = 3, endColumn = 15)
 
     val extracted = loc.extractFrom(input)
 
     assertThat(extracted).isEqualTo("""s Test {
           public String field
             = 'H""")
   }
+
+  @Test
+  fun trailingEmptyLine() {
+    val input = "public class Test { }\n"
+
+    val cu = TranslateHelpers.parseAndTranslate(input)
+    val loc = cu.getSourceLocation()
+
+    assertThat(loc.endLine).isEqualTo(2)
+    // There is no first column on the second, empty line.
+    // Expect zero in this case.
+    assertThat(loc.endColumn).isEqualTo(0)
```

Comment:
I guess this test clarifies to me a bit what you meant with the comment I ask above. Maybe I'm still missing some subtlety here... what happens if:

val input = "public class Test { }\n "

or

val input = "public class Test { }\n  "

?

Reply:
The test is testing when the end position is on an empty line because: what should the `endColumn` be when there are no columns? (I chose zero.)

That ambiguity goes away with your two cases. For #1, the `endColumn=0` will include all of the input characters. For #2, the `endColumn=1`.

Reply:
I added both tests.